src/TLuaInterpreter.cpp

```diff
@@ -15578,3 +15580,44 @@ int TLuaInterpreter::disableClickthrough(lua_State* L)
     mudlet::self()->setClickthrough(&host, windowName, false);
     return 0;
 }
+
+int TLuaInterpreter::getPlayerName(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    lua_pushstring(L, host.getLogin().toUtf8().constData());
+    return 1;
+}
+
+int TLuaInterpreter::sendPlayerName(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    QString name = host.getLogin();
+    if (name.isEmpty()) {
```

Comment:
Why this split? The `sendPlayerPassword()` below doesn't do it either?

Reply:
Split - do you mean why is sending the character name a separate function to sending the password?

If that is the question then this is because if you are scripting a login process triggered by prompts you will want to wait for each prompt (e.g. "login:" and then "password:" for a naked telnet server login for example) and respond to each of those separately.

If you mean - why store it in `name` I realise now that that is an oversight - I think I was going to do something else with the name and wanted a local copy because the calls to `Host` use a `QMutexLocker` and invoking that a second time here in the same function is non-optimum; as it is I will get rid of that local variable now.